plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/translate/SQLQueryTranslator.java

```diff
@@ -14,47 +14,182 @@
+import net.sf.jsqlparser.statement.create.table.Index;
 import org.jkiss.code.NotNull;
 import org.jkiss.dbeaver.DBException;
-import org.jkiss.dbeaver.model.preferences.DBPPreferenceStore;
 import org.jkiss.dbeaver.model.sql.SQLDialect;
+import org.jkiss.dbeaver.model.sql.SQLDialectDDLExtension;
+import org.jkiss.dbeaver.model.sql.SQLQuery;
 import org.jkiss.dbeaver.model.sql.SQLScriptElement;
-import org.jkiss.dbeaver.model.sql.parser.SQLScriptParser;
+import org.jkiss.dbeaver.model.sql.format.SQLFormatUtils;
+import org.jkiss.utils.CommonUtils;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.Locale;
 
 /**
- * SQL translator
+ * The type Sql query translator.
  */
-public final class SQLQueryTranslator {
-
-    public static String translateScript(
-        @NotNull SQLDialect sourceDialect,
-        @NotNull SQLDialect targetDialect,
-        @NotNull DBPPreferenceStore preferenceStore,
-        @NotNull String script) throws DBException
-    {
-        SQLTranslateContext context = new SQLTranslateContext(sourceDialect, targetDialect, preferenceStore);
-
-        List<SQLScriptElement> sqlScriptElements = SQLScriptParser.parseScript(sourceDialect, preferenceStore, script);
-        List<SQLScriptElement> result = new ArrayList<>();
-        for (SQLScriptElement element : sqlScriptElements) {
-            result.addAll(
-                context.translateCommand(element));
+public class SQLQueryTranslator implements SQLTranslator {
+
+    private SQLTranslateContext sqlTranslateContext;
+
+    public SQLQueryTranslator(SQLTranslateContext sqlTranslateContext) {
+        this.sqlTranslateContext = sqlTranslateContext;
+    }
+
+    public List<? extends SQLScriptElement> translate(@NotNull SQLScriptElement element) throws DBException {
+
+        if (element instanceof SQLQuery) {
+            return translateQuery((SQLQuery) element);
         }
-        String scriptDelimiter = targetDialect.getScriptDelimiters()[0];
 
-        StringBuilder sql = new StringBuilder();
-        for (SQLScriptElement element : result) {
-            sql.append(element.getText());
-            sql.append(scriptDelimiter).append("\n");
+        return Collections.singletonList(element);
+    }
+
+    private List<? extends SQLScriptElement> translateQuery(@NotNull SQLQuery query) {
+        Statement statement = query.getStatement();
+        if (statement != null) {
+            return translateStatement(query, statement);
         }
-        return sql.toString();
+        return Collections.singletonList(query);
     }
 
+    /**
+     * Translates statement to target dialect.
+     *
+     * @param query     the query
+     * @param statement the statement
+     * @return the list
+     */
+    protected List<? extends SQLScriptElement> translateStatement(@NotNull SQLQuery query,
+                                                                  @NotNull Statement statement) {
```

Comment:
Please specify nullability using an appropriate annotation

```suggestion
    protected List<? extends SQLScriptElement> translateStatement(
        @NotNull SQLQuery query,
        @NotNull Statement statement
    ) {
```

Reply:
fixed